### ISSUE TITLE
Fatal Error When Sending Campaign Emails with Attachments: Missing Request in CLI Execution

### ISSUE THREAD

**Issue by caiqueportela** [mautic/mautic#14305]
### Mautic Series

5.1.x series

### Mautic installed version

5.1.1

### Way of installing

I'm using a Docker image from https://hub.docker.com/r/mautic/mautic

### PHP version

8.2.25

### What browsers are you seeing the problem on?

Not relevant

### What happened?

In a campaign with an action to send an email, where the email includes an attachment, a fatal error occurs when trying to register the download tracking.

Analyzing the code, whenever an email with attachments is sent, a record is created for tracking the download. During this process, the **Request** is retrieved using the **RequestStack**. Since the campaign execution happens via CLI, there is no active HTTP request, leading to a fatal error.

As a result, the email is not sent.

### How can we reproduce this issue?

Step 1: Create a new asset.
Step 2: Create a campaign email template that includes the attachment.
Step 3: Create a form that collects the contact's email address.
Step 4: Create a campaign with the form as the source of contacts.
Step 5: Add an action to the campaign to send the email to the contact using the created template.
Step 6: Submit the form with a valid email address.
Step 7: When the campaign is processed (php bin/console mautic:campaigns:trigger), you will encounter the following error: `In AssetModel.php line 145: Call to a member function get() on null`

### Relevant log output

```shell
[2024-11-28T15:47:57.972595-03:00] console.CRITICAL: Error thrown while running command "mautic:campaigns:trigger". Message: "Call to a member function get() on null" {"exception":"[object] (Error(code: 0): Call to a member function get() on null at /var/www/html/docroot/app/bundles/AssetBundle/Model/AssetModel.php:145)","command":"mautic:campaigns:trigger","message":"Call to a member function get() on null"} {"hostname":"242cb9d4bcc8","pid":48}
[2024-11-28T15:47:57.973445-03:00] php.CRITICAL: Uncaught Error: Call to a member function get() on null {"exception":"[object] (Error(code: 0): Call to a member function get() on null at /var/www/html/docroot/app/bundles/AssetBundle/Model/AssetModel.php:145)"} {"hostname":"242cb9d4bcc8","pid":48}
[2024-11-28T15:47:57.972277-03:00] mautic.ERROR: Error: Call to a member function get() on null (uncaught exception) at /var/www/html/docroot/app/bundles/AssetBundle/Model/AssetModel.php line 145 while running console command `mautic:campaigns:trigger` [stack trace] #0 /var/www/html/docroot/app/bundles/EmailBundle/Helper/MailHelper.php(1578): Mautic\AssetBundle\Model\AssetModel->trackDownload(Object(Mautic\AssetBundle\Entity\Asset), NULL, 200, Array) #1 /var/www/html/docroot/app/bundles/EmailBundle/Helper/MailHelper.php(407): Mautic\EmailBundle\Helper\MailHelper->createAssetDownloadEntries() #2 /var/www/html/docroot/app/bundles/EmailBundle/Helper/MailHelper.php(460): Mautic\EmailBundle\Helper\MailHelper->send(false) #3 /var/www/html/docroot/app/bundles/EmailBundle/Model/SendEmailToContact.php(379): Mautic\EmailBundle\Helper\MailHelper->queue(false, 'RETURN_ERRORS') #4 /var/www/html/docroot/app/bundles/EmailBundle/Model/SendEmailToContact.php(190): Mautic\EmailBundle\Model\SendEmailToContact->sendStandardEmail() #5 /var/www/html/docroot/app/bundles/EmailBundle/Model/EmailModel.php(1451): Mautic\EmailBundle\Model\SendEmailToContact->send() #6 /var/www/html/docroot/app/bundles/EmailBundle/EventListener/CampaignSubscriber.php(313): Mautic\EmailBundle\Model\EmailModel->sendEmail(Object(Mautic\EmailBundle\Entity\Email), Array, Array) #7 /var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php(270): Mautic\EmailBundle\EventListener\CampaignSubscriber->onCampaignTriggerActionSendEmailToContact(Object(Mautic\CampaignBundle\Event\PendingEvent), 'mautic.email.on...', Object(Symfony\Component\EventDispatcher\EventDispatcher)) #8 /var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php(230): Symfony\Component\EventDispatcher\EventDispatcher::Symfony\Component\EventDispatcher\{closure}(Object(Mautic\CampaignBundle\Event\PendingEvent), 'mautic.email.on...', Object(Symfony\Component\EventDispatcher\EventDispatcher)) #9 /var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php(59): Symfony\Component\EventDispatcher\EventDispatcher->callListeners(Array, 'mautic.email.on...', Object(Mautic\CampaignBundle\Event\PendingEvent)) #10 /var/www/html/docroot/app/bundles/CampaignBundle/Executioner/Dispatcher/ActionDispatcher.php(43): Symfony\Component\EventDispatcher\EventDispatcher->dispatch(Object(Mautic\CampaignBundle\Event\PendingEvent), 'mautic.email.on...') #11 /var/www/html/docroot/app/bundles/CampaignBundle/Executioner/Event/ActionExecutioner.php(46): Mautic\CampaignBundle\Executioner\Dispatcher\ActionDispatcher->dispatchEvent(Object(Mautic\CampaignBundle\EventCollector\Accessor\Event\ActionAccessor), Object(Mautic\CampaignBundle\Entity\Event), Object(Doctrine\Common\Collections\ArrayCollection)) #12 /var/www/html/docroot/app/bundles/CampaignBundle/Executioner/EventExecutioner.php(127): Mautic\CampaignBundle\Executioner\Event\ActionExecutioner->execute(Object(Mautic\CampaignBundle\EventCollector\Accessor\Event\ActionAccessor), Object(Doctrine\Common\Collections\ArrayCollection)) #13 /var/www/html/docroot/app/bundles/CampaignBundle/Executioner/EventExecutioner.php(98): Mautic\CampaignBundle\Executioner\EventExecutioner->executeLogs(Object(Mautic\CampaignBundle\Entity\Event), Object(Doctrine\Common\Collections\ArrayCollection), Object(Mautic\CampaignBundle\Executioner\Result\Counter)) #14 /var/www/html/docroot/app/bundles/CampaignBundle/Executioner/EventExecutioner.php(169): Mautic\CampaignBundle\Executioner\EventExecutioner->executeForContacts(Object(Mautic\CampaignBundle\Entity\Event), Object(Doctrine\Common\Collections\ArrayCollection), Object(Mautic\CampaignBundle\Executioner\Result\Counter), false) #15 /var/www/html/docroot/app/bundles/CampaignBundle/Executioner/KickoffExecutioner.php(159): Mautic\CampaignBundle\Executioner\EventExecutioner->executeEventsForContacts(Object(Doctrine\Common\Collections\ArrayCollection), Object(Doctrine\Common\Collections\ArrayCollection), Object(Mautic\CampaignBundle\Executioner\Result\Counter)) #16 /var/www/html/docroot/app/bundles/CampaignBundle/Executioner/KickoffExecutioner.php(60): Mautic\CampaignBundle\Executioner\KickoffExecutioner->executeOrScheduleEvent() #17 /var/www/html/docroot/app/bundles/CampaignBundle/Command/TriggerCampaignCommand.php(332): Mautic\CampaignBundle\Executioner\KickoffExecutioner->execute(Object(Mautic\CampaignBundle\Entity\Campaign), Object(Mautic\CampaignBundle\Executioner\ContactFinder\Limiter\ContactLimiter), Object(Symfony\Component\Console\Output\ConsoleOutput)) #18 /var/www/html/docroot/app/bundles/CampaignBundle/Command/TriggerCampaignCommand.php(285): Mautic\CampaignBundle\Command\TriggerCampaignCommand->executeKickoff() #19 /var/www/html/docroot/app/bundles/CampaignBundle/Command/TriggerCampaignCommand.php(234): Mautic\CampaignBundle\Command\TriggerCampaignCommand->triggerCampaign(Object(Mautic\CampaignBundle\Entity\Campaign)) #20 /var/www/html/vendor/symfony/console/Command/Command.php(298): Mautic\CampaignBundle\Command\TriggerCampaignCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #21 /var/www/html/vendor/symfony/console/Application.php(1058): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #22 /var/www/html/vendor/symfony/framework-bundle/Console/Application.php(96): Symfony\Component\Console\Application->doRunCommand(Object(Mautic\CampaignBundle\Command\TriggerCampaignCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #23 /var/www/html/vendor/symfony/console/Application.php(301): Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand(Object(Mautic\CampaignBundle\Command\TriggerCampaignCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #24 /var/www/html/vendor/symfony/framework-bundle/Console/Application.php(82): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #25 /var/www/html/vendor/symfony/console/Application.php(171): Symfony\Bundle\FrameworkBundle\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #26 /var/www/html/bin/console(16): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #27 {main} [] {"hostname":"242cb9d4bcc8","pid":48}
[2024-11-28T15:47:57.972750-03:00] mautic.WARNING: Command `mautic:campaigns:trigger` exited with status code 1 [] {"hostname":"242cb9d4bcc8","pid":48}
```

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by matbcvo** [comment]
Thanks for reporting the bug, @caiqueportela
However, this appears to be a duplicate of an issue that has already been reported.
Please refer to #13577 for a detailed overview of the issue and the steps taken to resolve it.

**Comment by RCheesley** [comment]
Thanks @matbcvo for flagging up the duplicate! 🚀
